lib/puppet/provider/tacacs_server/cisco_nexus.rb

```diff
@@ -0,0 +1,74 @@
+require 'puppet/resource_api/simple_provider'
+
+# Implementation for the tacacs_server type using the Resource API.
+class Puppet::Provider::TacacsServer::CiscoNexus < Puppet::ResourceApi::SimpleProvider
+  def canonicalize(_context, resources)
+    resources
+  end
+
+  def get(_context, hosts=nil)
+    require 'cisco_node_utils'
+    current_states = []
+    @tacacs_server = Cisco::TacacsServerHost.hosts
```

Comment:
Reason not using `||=` is due to how creation works, the created host would not be available when handling it so need to reevaluate the `Cisco::TacacsServerHost.hosts` for recently created hosts

Reply:
Might be worth putting this as a code comment for future reference